lighthouse-core/scripts/release/test.sh

```diff
@@ -36,12 +36,14 @@ cd /tmp/lighthouse-local-test
 
 npm init -y
 npm install "$LH_PRISTINE_ROOT/lighthouse-$VERSION.tgz"
-npx add-dependencies package.json mime-types lodash.clonedeep
-npm install --only=prod
-cp -r "$LH_PRISTINE_ROOT/lighthouse-cli/test/fixtures" node_modules/lighthouse/lighthouse-cli/test
-npm explore lighthouse -- npm run smoke -- --tests-path "$LH_PRISTINE_ROOT/lighthouse-cli/test/smokehouse/test-definitions/core-tests.js" --retries=3
 npm explore lighthouse -- npm run fast -- http://example.com
 
+# Start up pristine's static-server (and kill it on exit).
+node "$LH_PRISTINE_ROOT/lighthouse-cli/test/fixtures/static-server.js" &
+trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT
```

Comment:
this is the only way I could get it to kill `static-server` since it's launching another process of its own (but it's not a child of it? I have no idea but it just wouldn't die) so e.g. saving the `node static-server` pid and then killing that wasn't sufficient. Happy to take suggestions from people who know much better what they're doing here, though :)

Reply:
this is the strongest argument I've seen yet for why we should just have an option allow the static server with `--fixture-path` :)

`--experimental-super-dangerous-server-dir` if we really don't want others to rely on its query params?

Reply:
> this is the strongest argument I've seen yet for why we should just have an option allow the static server with `--fixture-path` :)

well let's not go too wild based just on me not understanding how signals in process trees work :P I was hoping to nerd snipe someone into it, but @paulirish's on vacation and you and @connorjclark didn't take the bait :)

I moved up the static-server start to before the directory change, so at least that part can be the much cleaner `yarn static-server &`.

...that is an option though. Are we going to care much about this line after it lands until we next look at it in like 18 months from now, though? OTOH, `static-server` is definitely ready for a cleanup pass regardless of this (`fs.exists` has been deprecated since node 1.0!), and it might not be so bad to add at the same time.

Reply:
> not understanding how signals in process trees work

Haha, you're not alone here that's why I thought it was a strong argument to avoid dealing with them for all our sakes :)

I didn't jump in because I didn't think I understood the problem. I didn't see where static-server starts child processes, so I was surprised that this would be a thing. 

```
yarn static-server &
kill $!
```

worked just fine locally for me, so couldn't figure out where I could be sniped 🤷

Reply:
I think you have to give it a second to start up, so stick a `sleep 5` or something in between

Reply:
Hm, still fine.

```bash
cat > test.sh <<EOF
#!/bin/sh

yarn static-server &
sleep 5
curl http://localhost:10200
sleep 5
kill $!
curl http://localhost:10200
EOF
bash test.sh

Reply:
Yeah, I was trying something similar and see the same behavior with that script, two zombie node processes (I swear it was one before, but definitely two now, I assume one for each `http.Server`).

But looking into the process tree now, I think the problem might be that I'm using volta for managing node versions and the issue might be https://github.com/volta-cli/volta/issues/36, which is annoying :)

Maybe I'll leave the overkill `kill` but also leave a comment on it?

Reply:
(the last comment in that issue is what makes it seem most likely, because `ctrl-c` _does_ kill the node processes even without a `trap`)

Reply:
Ah, ok. Yeah I'd vote regular `kill` then 👍

Reply:
Well it's overkill for anyone not using volta, but it's not like it's doing anything bad, everything should still work as expected. It's just killing its own process group so all child processes are taken out instead of a child process + any of _its_ children. My understanding is the first part is just to [prevent trap recursion in recent bash](https://stackoverflow.com/questions/360201/how-do-i-kill-background-processes-jobs-when-my-shell-script-exits/28333938#28333938).